azurelinuxagent/ga/exthandlers.py

```diff
@@ -150,8 +150,13 @@ def parse_ext_substatus(substatus):
 
 
 def parse_ext_status(ext_status, data):
-    if data is None or len(data) == 0: # pylint: disable=len-as-condition
+    if data is None:
         return
+    if not isinstance(data, list):
+        raise ExtensionStatusError(msg="The extension status must be an array: {0}".format(data), code=ExtensionStatusError.StatusFileMalformed)
```

Comment:
Data would contain the complete content of the status file, wont that be too verbose if the extension status is huge (multiple substatuses, etc)?

Reply:
Yes, I thought about truncating it, but I really want to capture the extension name if it is there. The verbosity wouldn't matter too much, I think. I can always truncate it to 1 or 2 K.

Reply:
actually, the error message also goes to telemetry... let me check if we truncate or if a large message would drop the event

Reply:
yes, greater that 64512 we would drop the event. i'll truncate at 4K